fdbserver/RESTSimKmsVault.actor.cpp

```diff
@@ -0,0 +1,597 @@
+		Reference<SimKmsVaultKeyCtx> keyCtx = SimKmsVault::getByDomainId(domainId);
+		ASSERT(keyCtx.isValid());
+		addCipherDetailToRespDoc(result, cipherDetails, keyCtx, domainId);
+	}
+	rapidjson::Value memberKey(CIPHER_KEY_DETAILS_TAG, result.GetAllocator());
+	result.AddMember(memberKey, cipherDetails, result.GetAllocator());
+
+	if (doc.HasMember(KMS_URLS_TAG) && doc[KMS_URLS_TAG].GetBool()) {
+		addKmsUrlsToDoc(result);
+	}
+
+	// Serialize json to string
+	rapidjson::StringBuffer sb;
+	rapidjson::Writer<rapidjson::StringBuffer> writer(sb);
+	result.Accept(writer);
+
+	ASSERT(!response.failed);
+	response.buff = std::string(sb.GetString(), sb.GetSize());
+	//TraceEvent("ResponeStr").detail("Str", response->buff);
+	return response;
+}
+
+VaultResponse handleFetchKeysByKeyIds(const std::string& content) {
+	VaultResponse response;
+	rapidjson::Document doc;
+
+	doc.Parse(content.data());
+
+	int version;
+
+	if (!extractVersion(doc, &response, &version)) {
+		// Return HTTP::HTTP_STATUS_CODE_OK with appropriate 'error' details
+		ASSERT(response.failed);
+		return response;
+	}
+	ASSERT(!response.failed);
+
+	if (!checkValidationTokens(doc, version, &response)) {
+		// Return HTTP::HTTP_STATUS_CODE_OK with appropriate 'error' details
+		ASSERT(response.failed);
+		return response;
+	}
+	ASSERT(!response.failed);
+
+	rapidjson::Document result;
+	result.SetObject();
+
+	// Append 'request version'
+	addVersionToDoc(result, version);
+
+	// Append 'cipher_key_details' as json array
+	rapidjson::Value cipherDetails(rapidjson::kArrayType);
+	for (const auto& cipherDetail : doc[CIPHER_KEY_DETAILS_TAG].GetArray()) {
+		Optional<EncryptCipherDomainId> domainId;
+		if (cipherDetail.HasMember(ENCRYPT_DOMAIN_ID_TAG) && cipherDetail[ENCRYPT_DOMAIN_ID_TAG].IsInt64()) {
+			domainId = cipherDetail[ENCRYPT_DOMAIN_ID_TAG].GetInt64();
+		}
+		EncryptCipherBaseKeyId baseCipherId = cipherDetail[BASE_CIPHER_ID_TAG].GetUint64();
+		Reference<SimKmsVaultKeyCtx> keyCtx = SimKmsVault::getByBaseCipherId(baseCipherId);
+		ASSERT(keyCtx.isValid());
+		addCipherDetailToRespDoc(result, cipherDetails, keyCtx, domainId);
+	}
+	rapidjson::Value memberKey(CIPHER_KEY_DETAILS_TAG, result.GetAllocator());
+	result.AddMember(memberKey, cipherDetails, result.GetAllocator());
+
+	if (doc.HasMember(KMS_URLS_TAG) && doc[KMS_URLS_TAG].GetBool()) {
+		addKmsUrlsToDoc(result);
+	}
+
+	// Serialize json to string
+	rapidjson::StringBuffer sb;
+	rapidjson::Writer<rapidjson::StringBuffer> writer(sb);
+	result.Accept(writer);
+
+	ASSERT(!response.failed);
+	response.buff = std::string(sb.GetString(), sb.GetSize());
+	TraceEvent("ResponeStr").detail("Str", response.buff);
+	return response;
+}
+
+ACTOR Future<Void> simKmsVaultRequestHandler(Reference<HTTP::IncomingRequest> request,
+                                             Reference<HTTP::OutgoingResponse> response) {
+	wait(delay(0));
+	ASSERT_EQ(request->verb, HTTP::HTTP_VERB_POST);
```

Comment:
doesn't have to be this PR but we might also want to double check the other headers (for example that it is requesting content type json, and we might want to have the response specify via header that it is using json as well.

Reply:
Good point. I will update it in the upcoming PR.